app/assets/javascripts/govuk_publishing_components/components/search-with-autocomplete.js

```diff
@@ -0,0 +1,99 @@
+      this.$form = this.$module.closest('form')
+
+      this.source = this.$module.getAttribute('data-source')
+      this.sourceKey = this.$module.getAttribute('data-source-key')
+    }
+
+    init () {
+      // Do not enhance the input field if we won't be able to provide autocomplete suggestions
+      if (!this.source || !this.sourceKey) {
+        console.warn('search-with-autocomplete: No source/sourceKey provided')
+        return
+      }
+
+      const configOptions = {
+        element: this.$inputWrapper,
+        id: this.$originalInput.id,
+        name: this.$originalInput.name,
+        inputClasses: this.$originalInput.classList,
+        defaultValue: this.$originalInput.value,
+        confirmOnBlur: false,
+        showNoOptionsFound: false,
+        source: this.getResults.bind(this),
+        onConfirm: this.submitContainingForm.bind(this),
+        templates: {
+          suggestion: this.constructSuggestionHTMLString.bind(this)
+        }
+      }
+      accessibleAutocomplete(configOptions)
+
+      // The accessible-autocomplete component is meant to generate a new input element rather than enhancing an existing one, so we need to do some cleanup here.
+      const autocompleteInput = this.$inputWrapper.querySelector('.autocomplete__input')
+      // Ensure the new input element generated by accessible-autocomplete has the correct type
+      autocompleteInput.setAttribute('type', 'search')
+      // Remove the original input from the DOM
+      this.$originalInput.parentNode.removeChild(this.$originalInput)
+    }
+
+    // Callback used by accessible-autocomplete to generate the HTML for each suggestion based on
+    // the values returned from the source
+    constructSuggestionHTMLString (result) {
+      // We should be able to assume that data coming back from the Autocomplete API is safe to
+      // render as HTML. However, this is a final line of defence to prevent XSS.
+      const sanitizedResult = result.replace(/[&<>"']/g, '')
+      const inputValue = this.$inputWrapper.querySelector('input').value.toLowerCase()
+
+      const index = sanitizedResult.toLowerCase().indexOf(inputValue)
+
+      let html = sanitizedResult
+      if (index !== -1) {
+        const before = sanitizedResult.slice(0, index)
+        const match = sanitizedResult.slice(index, index + inputValue.length)
+        const after = sanitizedResult.slice(index + inputValue.length)
+
+        html = `${before}<mark>${match}</mark>${after}`
+      }
+
+      return `
+        <div class="autocomplete__option-wrapper">
+          <span class="autocomplete__suggestion-icon"></span>
+          <span class="autocomplete__suggestion-text">${html}</span>
+        </div>
+      `
+    }
+
+    // Callback used by accessible-autocomplete to fetch results from the source
+    getResults (query, populateResults) {
+      const url = new URL(this.source)
+      url.searchParams.set('q', query)
+      fetch(url)
```

Comment:
should we specifying that we want a response of JSON or are we safe to assume the endpoints will only ever serve JSON?